include/wlr/types/wlr_region.h

```diff
@@ -0,0 +1,10 @@
+#ifndef _WLR_TYPES_REGION_H
+#define _WLR_TYPES_REGION_H
+
+struct wl_resource;
+
+// Implements the given resource as region.
+// Sets the associated pixman_region32_t as userdata.
+void wlr_region_create(struct wl_resource *res);
```

Comment:
thought about, are you sure this is a good idea? the region is directly connected with the resource so why should it be needed on creation? changing it independently from the client will result in weird behavior and it should only be read when it is passed by the client as parameter somewhere i guess